File: ParcelablesPrac/app/src/main/java/com/example/bob/parcelablesprac/MainActivity.java
package com.example.bob.parcelablesprac;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Wow, that's pretty darn neat!
        Intent toSecond = new Intent(getApplicationContext(), SecondActivity.class);
        toSecond.putExtra("student", new Student(1,"Mike","6"));
        startActivity(toSecond);
    }
}

File: TestLayout/app/src/main/java/com/example/bob/testlayout/MainActivity.kt
package com.example.bob.testlayout

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v4.app.Fragment

//https://stackoverflow.com/questions/23149981/fragments-the-specified-child-already-has-a-parent-you-must-call-removeview
//Fixed the mentioned problem via above link.
class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        supportFragmentManager.beginTransaction()
                .add(R.id.frag_container_one, FragOne(), "frag_one")
                .add(R.id.frag_container_two, FragTwo(), "frag_two")
                .commit()
    }
}